opencanary/logger.py

```diff
@@ -334,8 +335,11 @@ def emit(self, record):
         if self.data is None:
             data = mapping
         else:
-            # Due to the recursive function, sending a shallow copy of self.data here is fine.
-            data = map_string(dict(self.data), mapping)
+            if isinstance(self.data, dict):
```

Comment:
This doesn't follow for me. It looks like we are saying that if `self.data` is a `dict`, we should cast it to a `dict` - which feels unnecessary

Reply:
I'd be happy to explain. Essentially, `self.data` is not a dict [but rather a `logging.config.ConvertingDict`](https://github.com/python/cpython/blob/main/Lib/logging/config.py#L340) - which is a sub-class of dict. Using `isinstance(self.data, dict)` in this case is a catch-all for any dict or sub-class of dict. I could have imported the class and used `isinstance(self.data, ConvertingDict)`, but I took a shortcut.

As for the second part, we need to cast it to a standard dict so that when it passes through the recursive function `map_string` we use the standard `__getitem__`, `.get()` and `.pop()` methods instead of the ones from ConvertingDict ([which try and convert the keys](https://github.com/python/cpython/blob/main/Lib/logging/config.py#L310) and was not working when I tested it).

Reply:
Awesome thanks for explanation @joewesch 🤘 that clears it up for silly me haha.